sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -22,7 +25,16 @@ public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNa
             IsActivityEnabled = isActivityEnabled;
             if (IsActivityEnabled)
             {
-                _source = new DiagnosticListener(clientNamespace);
+                LazyInitializer.EnsureInitialized(ref _listeners);
```

Comment:
PERF: (minor) Note that LazyInitializer.EnsureInitialized returns the initialized value. This is useful because you can assign that to a local variable and avoid accessing the _listeners field again. That should open up slightly better code gen.